File: test_game.js
const { chromium } = require('playwright');
(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage();
  page.on('console', msg => console.log('PAGE LOG:', msg.text()));
  page.on('pageerror', error => console.log('PAGE ERROR:', error.message));
  page.on('response', response => {
    if (response.status() === 404) console.log('404 URL:', response.url());
  });
  await page.goto('http://localhost:8083/Assets/bad-ice-cream-2/index.html', { waitUntil: 'networkidle' });
  await page.waitForTimeout(5000); // Wait for loading
  await browser.close();
})();

File: test_game.py
from playwright.sync_api import sync_playwright
import time

def main():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()
        page.on("console", lambda msg: print(f"CONSOLE {msg.type}: {msg.text}"))
        print("Navigating...")
        page.goto("http://localhost:8086/bad-ice-cream-3/")
        time.sleep(10)
        
        # Check splash visibility
        splash_info = page.evaluate('''() => {
            const splash = document.getElementById('splash__image');
            return splash ? window.getComputedStyle(splash).visibility : "No splash";
        }''')
        print("Splash visibility:", splash_info)
        
        browser.close()

main()
